tests/test_alerts.py

```diff
@@ -496,19 +542,197 @@ def test_cli_alerts_search_when_default_params_makes_expected_api_call(
+                "filters": [
+                    {
+                        "term": "Name",
+                        "operator": "IS",
+                        "value": "foo",
+                    },
+                ],
+            },
+            {
+                "filterClause": "AND",
+                "filters": [
+                    {
+                        "term": "Actor",
+                        "operator": "IS",
+                        "value": "foo@bar.com",
+                    },
+                ],
+            },
+            {
+                "filterClause": "AND",
+                "filters": [
+                    {
+                        "term": "ActorId",
+                        "operator": "IS",
+                        "value": "foo-42",
+                    },
+                ],
+            },
+            {
+                "filterClause": "AND",
+                "filters": [
+                    {
+                        "term": "RiskSeverity",
+                        "operator": "IS",
+                        "value": "LOW",
+                    },
+                ],
+            },
+            {
+                "filterClause": "AND",
+                "filters": [
+                    {
+                        "term": "State",
+                        "operator": "IS",
+                        "value": "OPEN",
+                    },
+                ],
+            },
+            {
+                "filterClause": "AND",
+                "filters": [
+                    {
+                        "term": "RuleId",
+                        "operator": "IS",
+                        "value": "baz",
+                    },
+                ],
+            },
+            {
+                "filterClause": "AND",
+                "filters": [
+                    {
+                        "term": "AlertSeverity",
+                        "operator": "IS",
+                        "value": "HIGH",
+                    },
+                ],
+            },
+        ],
+        "pgNum": 0,
+        "pgSize": 100,
+        "srtDirection": "DESC",
+        "srtKey": "CreatedAt",
+    }
+
+    httpserver_auth.expect_request(
+        "/v1/alerts/query-alerts", method="POST", json=query
+    ).respond_with_json(TEST_ALERTS_RESPONSE)
+
+    result = runner.invoke(
+        incydr,
+        [
+            "alerts",
+            "search",
+            "--start",
+            "2022-06-01",
+            "--end",
+            "2022-07-02",
+            "--alert-id",
+            "bar",
+            "--type",
+            "alert",
+            "--name",
+            "foo",
+            "--actor",
+            "foo@bar.com",
+            "--actor-id",
+            "foo-42",
+            "--risk-severity",
+            "LOW",
+            "--state",
+            "OPEN",
+            "--rule-id",
+            "baz",
+            "--alert-severity",
+            "HIGH",
+        ],
+    )
+    httpserver_auth.check()
```

Comment:
I discovered that this doesn't actually check that requests were made...

The `.expect_request()` method is "permanent" so can be hit multiple times in any order without raising an error. And `.check()` does something weird with assertions added to the response handlers that I don't fully understand.

The `.expect_ordered_request()` and `.expect_oneshot_request()` calls are both added to lists on the `HTTPServer` instance and removed once the request comes through. So to check that our expected requests were actually called we'd need to do:

```
httpserver_auth.expect_ordered_request(
        "/v1/alerts/query-alerts", method="POST", json=query
    ).respond_with_json(TEST_ALERTS_RESPONSE)
```
then:
```
assert len(httpserver_auth.ordered_handlers) == 0
```

Reply:
I know it doesn't check that requests were made, I use the `httpserver_auth.check()` method because I think I'd run into a previous issue where errors thrown by the SDK won't fail the test if the command doesn't use the response (like some of the update commands).  The pattern you suggested would be a good way to check that specific requests are made though, but I'm not sure it's worth rewriting all of our tests in this PR?

Reply:
Yeah, no need to update them in this PR, I just recently was surprised by `.check()` not doing what I thought it was doing.

Reply:
Oh yeah, its a little misleading. I think it basically just checks that any calls made were successful and that the httpserver client itself hasn't raised any errors 

> If you run the above code you will notice that no test failed. This is because the http server is running in its own thread, separately from the client code. It has no way to raise an assertion error in the client thread.

from https://pytest-httpserver.readthedocs.io/en/latest/tutorial.html#oneshot-handling

Reply:
Ah, yeah I missed the threading piece, makes sense now.